ocw_data_parser/static_html_generator.py

```diff
@@ -37,43 +38,51 @@ def generate_index_page(mj, destination):
+    s = "<!DOCTYPE html>\n<html lang=\"en\">\n<head>\n<meta charset=\"UTF-8\">\n<title>%s</title>\n<link rel=\"stylesheet\" href=\"https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css\" integrity=\"sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T\" crossorigin=\"anonymous\"><script src=\"https://code.jquery.com/jquery-3.3.1.slim.min.js\" integrity=\"sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo\" crossorigin=\"anonymous\"></script><script src=\"https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js\" integrity=\"sha384-UO2eT0CpHqdSJQ6hJty5KVphtPhzWj9WO1clHTMGa3JDZwrnQq4sF86dIHNDz0W1\" crossorigin=\"anonymous\"></script><script src=\"https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js\" integrity=\"sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM\" crossorigin=\"anonymous\"></script>\n</head>\n<body style=\"background-color: floralwhite\">\n" \
         % mj["title"]
     return s
 
 
 def generate_breadcrumbs(mj, current_page_uid):
     s = "<nav><ol class=\"breadcrumb\">\n%s</ol>\n</nav>\n"
     page_trail = list()
+    embedded_media = None
+    # Find out if we are viewing some embedded media
+    if "course_embedded_media" in mj:
+        for inline_embed_id in mj['course_embedded_media']:
+            media = mj['course_embedded_media'][inline_embed_id]
+            if current_page_uid == media["uid"]:
+                embedded_media = media
     # Find which page we are on
     for page in mj["course_pages"]:
         if page["uid"] == current_page_uid:
             page_trail.append(page)
             break
+        elif embedded_media:
+            if page["uid"] == embedded_media["parent_uid"]:
+                page_trail.append(page)
+                page_trail.append(embedded_media)
+                break
     # If page_trail is empty that means we're generating breadcrumbs for the home page
     if not page_trail:
         breadcrumbs = "<li class=\"breadcrumb-item active\"><a href=\"index.html\">%s</a></li>\n" % mj["title"]
-    else:
-        while page_trail[-1]["parent_uid"] != mj["uid"]:
-            for parent_page in mj["course_pages"]:
-                if parent_page["uid"] == page_trail[-1]["parent_uid"]:
-                    page_trail.append(parent_page)
-                    break
-        page_trail.reverse()
-        breadcrumbs = "<li class=\"breadcrumb-item\"><a href=\"index.html\">%s</a></li>\n" % mj["title"]
+    if page_trail:
```

Comment:
`else:`?

Reply:
Right you are.  I forget why I explicitly made a second if statement here, but it should definitely just be an else.